pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/TableConfigsRestletResource.java

```diff
@@ -0,0 +1,358 @@
+      validateConfig(tableConfigs);
+    } catch (Exception e) {
+      throw new ControllerApplicationException(LOGGER, String.format("Invalid TableConfigs. %s", e.getMessage()),
+          Response.Status.BAD_REQUEST, e);
+    }
+
+    TableConfig offlineTableConfig = tableConfigs.getOffline();
+    TableConfig realtimeTableConfig = tableConfigs.getRealtime();
+    Schema schema = tableConfigs.getSchema();
+
+    try {
+      String endpointUrl = request.getRequestURL().toString();
+      validatePermissions(schema.getSchemaName(), AccessType.CREATE, httpHeaders, endpointUrl);
+
+      if (offlineTableConfig != null) {
+        tuneConfig(offlineTableConfig, schema);
+        validatePermissions(offlineTableConfig.getTableName(), AccessType.CREATE, httpHeaders, endpointUrl);
+      }
+      if (realtimeTableConfig != null) {
+        tuneConfig(realtimeTableConfig, schema);
+        validatePermissions(realtimeTableConfig.getTableName(), AccessType.CREATE, httpHeaders, endpointUrl);
+      }
+
+      _pinotHelixResourceManager.addSchema(schema, true);
+      LOGGER.info("Added schema: {}", schema.getSchemaName());
+      if (offlineTableConfig != null) {
+        _pinotHelixResourceManager.addTable(offlineTableConfig);
+        LOGGER.info("Added offline table config: {}", offlineTableConfig.getTableName());
+      }
+      if (realtimeTableConfig != null) {
+        _pinotHelixResourceManager.addTable(realtimeTableConfig);
+        LOGGER.info("Added realtime table config: {}", realtimeTableConfig.getTableName());
+      }
+
+      return new SuccessResponse("TableConfigs " + tableConfigs.getTableName() + " successfully added");
+    } catch (Exception e) {
+      _controllerMetrics.addMeteredGlobalValue(ControllerMeter.CONTROLLER_TABLE_ADD_ERROR, 1L);
+      if (e instanceof PinotHelixResourceManager.InvalidTableConfigException) {
+        throw new ControllerApplicationException(LOGGER,
+            String.format("Invalid TableConfigs: %s", tableConfigs.getTableName()), Response.Status.BAD_REQUEST, e);
+      } else if (e instanceof PinotHelixResourceManager.TableAlreadyExistsException) {
+        throw new ControllerApplicationException(LOGGER, e.getMessage(), Response.Status.CONFLICT, e);
+      } else {
+        throw new ControllerApplicationException(LOGGER, e.getMessage(), Response.Status.INTERNAL_SERVER_ERROR, e);
+      }
+    }
+  }
+
+  /**
+   * Deletes the {@link TableConfigs} by deleting the schema tableName, the offline table config for tableName_OFFLINE and
+   * the realtime table config for tableName_REALTIME
+   */
+  @DELETE
+  @Path("/tableConfigs/{tableName}")
+  @Authenticate(AccessType.DELETE)
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Delete the TableConfigs", notes = "Delete the TableConfigs")
+  public SuccessResponse deleteConfig(
+      @ApiParam(value = "TableConfigs name i.e. raw table name", required = true) @PathParam("tableName") String tableName) {
+
+    try {
```

Comment:
Check table existence before deletion

Reply:
I am trying to keep it similar to the PinotTableRestletResource, where we call delete whether or not the table exists. I'm guessing it's because delete has multiple steps underneath and we won't know if table got deleted completely by checking `pinotHelixResourceManager.hasOfflineTable(tableName)`.